From 06c1debd468355266cdca9fdd96c9e0c74f24bba Mon Sep 17 00:00:00 2001
From: ksooo <kai.sommerfeld@gmx.com>
Date: Sun, 27 Sep 2026 20:53:43 +0200
Subject: [PATCH] Return the PCM buffer size from
 projectm_pcm_get_max_samples()

projectm_pcm_get_max_samples() is documented to return the number of
samples stored per channel, but returned WaveformSamples (480) instead
of AudioBufferSamples (576). Applications that limit the samples they
pass per frame to this value leave the oldest part of the analyzed
buffer filled with stale data, which causes visible jumps in the
waveform.
---
 src/libprojectM/ProjectMCWrapper.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libprojectM/ProjectMCWrapper.cpp b/src/libprojectM/ProjectMCWrapper.cpp
index cd2bca2..911b042 100644
--- a/src/libprojectM/ProjectMCWrapper.cpp
+++ b/src/libprojectM/ProjectMCWrapper.cpp
@@ -453,7 +453,7 @@ bool projectm_get_preset_start_clean(projectm_handle instance)
 
 unsigned int projectm_pcm_get_max_samples()
 {
-    return libprojectM::Audio::WaveformSamples;
+    return libprojectM::Audio::AudioBufferSamples;
 }
 
 template<class BufferType>
-- 
2.54.0 (Apple Git-157)

